### ISSUE TITLE
Type `inheritedTypes` should flatten types hierarchy

### ISSUE THREAD

**Issue by ilyapuchka** [krzysztofzablocki/Sourcery#41]
Currently we only put into `inheritedTypes` types that we directly inherit from or implement. Shouldn't we flatten types hierarchy so that if type that implements a protocol (or inherits from class) that inherits from another protocol we can check in the template if type implements both of them?

For example with such code:
```swift
protocol AutoEquatable: Equatable {}
class BaseClass: AutoEquatable {}
class SubClass: BaseClass {}
```

in the template `{% types.implementing.Equatable %}`will not give us any type when it should give us both `BaseClass` and `SubClass`. `{% types.implementing.AutoEquatable %}` will only give us `BaseClass`, not `SubClass`.

Implementation should be trivial I think.

**Comment by krzysztofzablocki** [comment]
:+1: I'll modify it

**Comment by krzysztofzablocki** [comment]
Implemented in 0.4.1, @bolismauro Can you give it a try?
